### DISCUSSION TITLE
请各位帮我看下这个使用了fallbacks和使用xhttp的配置文件有什么问题，不通

### DISCUSSION THREAD

**Discussion by c21xdx** [XTLS/Xray-core#4134] [General]
根路径是vless+ws,分了2条路径分别到vless+xhttp和vmess+ws,vless+ws和vmess+ws是通的，vless+xhttp不通。用的docker部署，没有前置nginx或caddy，直接xray运行，xray是最新版支持xhttp的。大家帮看看是我哪里搞错了

> {
	"log": {
		"access": "/dev/null",
		"error": "/dev/null",
		"loglevel": "debug"
	},
	"inbounds": [{
			"port": 8080,
			"protocol": "vless",
			"settings": {
				"clients": [{
					"id": "545727e7-cdb9-485c-8a80-ee61ad8c7db7"
				}],
				"decryption": "none",
				"fallbacks": [{
						"dest": 3001
					},
					{
						"path": "/xhttp",
						"dest": 3002
					},
					{
						"path": "/vmess",
						"dest": 3003
					}
				]
			},
			"streamSettings": {
				"network": "tcp"
			}
		},
		{
			"port": 3001,
			"listen": "127.0.0.1",
			"protocol": "vless",
			"settings": {
				"clients": [{
					"id": "545727e7-cdb9-485c-8a80-ee61ad8c7db7"
				}],
				"decryption": "none"
			},
			"streamSettings": {
				"network": "ws",
				"security": "none"
			}
		},
		{
			"port": 3002,
			"listen": "127.0.0.1",
			"protocol": "vless",
			"settings": {
				"clients": [{
					"id": "545727e7-cdb9-485c-8a80-ee61ad8c7db7",
					"level": 0
				}],
				"decryption": "none"
			},
			"streamSettings": {
				"network": "xhttp",
				"security": "none",
				"xhttpSettings": {
					"path": "/xhttp"
				}
			},
			"sniffing": {
				"enabled": true,
				"destOverride": [
					"http",
					"tls"
				],
				"metadataOnly": false
			}
		},
		{
			"port": 3003,
			"listen": "127.0.0.1",
			"protocol": "vmess",
			"settings": {
				"clients": [{
					"id": "545727e7-cdb9-485c-8a80-ee61ad8c7db7",
					"alterId": 0
				}]
			},
			"streamSettings": {
				"network": "ws",
				"wsSettings": {
					"path": "/vmess"
				}
			},
			"sniffing": {
				"enabled": true,
				"destOverride": [
					"http",
					"tls"
				],
				"metadataOnly": false
			}
		}
	],
	"routing": {
		"rules": [{
			"type": "field",
			"protocol": [
				"bittorrent"
			],
			"outboundTag": "block"
		}]
	},
	"dns": {
		"servers": [
			"https+local://8.8.8.8/dns-query"
		]
	},
	"outbounds": [{
			"protocol": "freedom"
		},
		{
			"protocol": "blackhole",
			"tag": "block"
		}
	]
}

**Comment by Benjamin1919** [comment]
Fallback 是TCP层级的回落，只有基于 http/1.1 的 websocket 和 httpupgrade 才能从首包中读出 path，其他传输层（比如 grpc、xhttp）是无法读取 path 的，因此不能用 path 回落。
  你这种情况应该删除 xhttp 的 path 匹配，并把它放在第一个回落。剩下的几个回落必须都带 path

**Reply by lxhao61** [reply]
他整个配置架构就错了，VLESS+TCP 可没有回落分流功能。

**Comment by lxhao61** [comment]
@c21xdx 你提供配置问题很多！不是简单说一下就可改正确的，建议参考 [Xray(E+H+G+B+A)+Caddy](https://github.com/lxhao61/integrated-examples/tree/main/Xray(E%2BH%2BG%2BB%2BA)%2BCaddy) 或 [Xray(E+H+G+B+A)+Nginx](https://github.com/lxhao61/integrated-examples/tree/main/Xray(E%2BH%2BG%2BB%2BA)%2BNginx) 修改即可。
